src/app/utils/BountyUtils.ts

```diff
@@ -420,20 +420,29 @@ const BountyUtils = {
         if (isDraftBounty) {  // If we are in Create (Draft) mode, put the card in the DM channel
             cardMessage = await (await DiscordUtils.getGuildMemberFromUserId(bounty.createdBy.discordId, bounty.customerId)).send(cardEmbeds);
         } else {
-            if (activity == Activities.publish) {  // Publishing. If the card exists, delete it - it was in a DM}
-                if (!!bounty.canonicalCard) {
-                    const draftChannel = await DiscordUtils.getTextChannelfromChannelId(bounty.canonicalCard.channelId);
-                    const draftCardMessage = await DiscordUtils.getMessagefromMessageId(bounty.canonicalCard.messageId, draftChannel);
-                    await draftCardMessage.delete();
-                    bounty.canonicalCard = undefined;
+            if (activity == Activities.publish) {  // Publishing. If the card exists, delete it - it was either in a DM or needs to be refreshed
+                if (!!bounty.canonicalCard) { 
+                    try {
+                        const draftChannel = await DiscordUtils.getTextChannelfromChannelId(bounty.canonicalCard.channelId);
+                        const draftCardMessage = await DiscordUtils.getMessagefromMessageId(bounty.canonicalCard.messageId, draftChannel);
+                        await draftCardMessage.delete();
+                    } catch(e) {
+                    } finally {
+                        bounty.canonicalCard = undefined;
+                    }
                 }
             }
-            if (!!bounty.canonicalCard) {  // If we still have an existing card, just edit it, remove old reactions
-                bountyChannel = await DiscordUtils.getTextChannelfromChannelId(bounty.canonicalCard.channelId);
-                cardMessage = await DiscordUtils.getMessagefromMessageId(bounty.canonicalCard.messageId, bountyChannel);
-                await cardMessage.edit(cardEmbeds);
-                await cardMessage.reactions.removeAll();
-            } else {  // Otherwise create it. Put it in the passed in channel, or customer channel by default
+            if (!!bounty.canonicalCard) {  // If we still have an existing card, try to just edit it, remove old reactions
+                try {
+                    bountyChannel = await DiscordUtils.getTextChannelfromChannelId(bounty.canonicalCard.channelId);
+                    cardMessage = await DiscordUtils.getMessagefromMessageId(bounty.canonicalCard.messageId, bountyChannel);
+                    await cardMessage.edit(cardEmbeds);
+                    await cardMessage.reactions.removeAll();
+                } catch(e) {
+                    bounty.canonicalCard = undefined;
+                }
+            } 
+            if (!bounty.canonicalCard) { // If we didn't have a card, or we had an error trying to access it, create it
```

Comment:
minor, but I might mark this as an `else if` for readability, though I think it's more readable than a simple `else`

Reply:
We can't do an else here, because if one of the try-catches above detects a missing card or channel, we need to rebuild the card.

Reply:
I don't think that effects the behavior of the if statement, as `if (!!bounty.canonicalCard) {` and `if (!bounty.canonicalCard)` are at the same level in the logical flow.

Reply:
Maybe in misunderstanding. You want an else in front of the second if? Then in bounty.canonicalCard is set to undefined in one of the previous catches, the else would prevent the card from being rebuilt, right?